### DISCUSSION TITLE
Is it possible to create state actions?

### DISCUSSION THREAD

**Discussion by marcebdev** [pmndrs/jotai#2381] [Q&A]
By state actions I mean similar to how virtually all flux state stores operate, but something that I think is very critical to what any global state store is: being able to pull logic outside of React where it makes sense.

One use case is very similar to this: https://github.com/pmndrs/jotai/discussions/1752 i.e. you don't need to run the network requests in React itself and you just want an action to set update some state based on the response, you don't need React to be involved.

But in general, as you make more complex stores the concept of actions itself become critically important to organization, and I'd argue towards helpful towards reducing re-renders as well (why always mount to React if you have operations that have no reason to touch a particular component tree). It seems that in order to interact with stores you still either need to subscribe to the state in a component, or you can `createStore()`. However, you then need to use a `Provider` which no longer makes it global state and there's not really any recommended patterns around it.

So I wanted to see if there's patterns I'm missing for `action` like functionality.

I'd love to use Jotai more, and with V2 I wanted to give it another look for a new project so I'm hoping there's an option I'm missing. I'd like to achieve functionality similar to how you can use [nanostores](https://github.com/nanostores/nanostores).

**Comment by dai-shi** [comment]
I hope someone can comment on "actions", but apart from "actions", it seems you might be missing some features like `useSetAtom` and `getDefaultStore`.
